usaspending_api/common/sqs/sqs_work_dispatcher.py

```diff
@@ -0,0 +1,990 @@
+import logging
+import inspect
+import json
+import os
+
+import psutil as ps
+import signal
+import time
+import multiprocessing as mp
+
+from botocore.exceptions import ClientError, EndpointConnectionError, NoCredentialsError, NoRegionError
+
+from usaspending_api.common.sqs.sqs_handler import get_sqs_queue
+from usaspending_api.common.sqs.sqs_job_logging import log_job_message
+
+# Not a complete list of signals
+# NOTE: 1-15 are relatively standard on unix platforms; > 15 can change from platform to platform
+BSD_SIGNALS = {
+    1: "SIGHUP [1] (Hangup detected on controlling terminal or death of controlling process)",
+    2: "SIGINT [2] (Interrupt from keyboard)",
+    3: "SIGQUIT [3] (Quit from keyboard)",
+    4: "SIGILL [4] (Illegal Instruction)",
+    6: "SIGABRT [6] (Abort signal from abort(3))",
+    8: "SIGFPE [8] (Floating point exception)",
+    9: "SIGKILL [9] (Non-catchable, non-ignorable kill)",
+    11: "SIGSEGV [11] (Invalid memory reference)",
+    14: "SIGALRM [14] (Timer signal from alarm(2))",
+    15: "SIGTERM [15] (Software termination signal)",
+    19: "SIGSTOP [19] (Suspend process execution)",  # NOTE: 17 on Mac OSX, 19 on RHEL
+    20: "SIGTSTP [20] (Interrupt from keyboard to suspend (CTRL-Z)",  # NOTE: 18 on Mac OSX, 20 on RHEL
+}
+
+
+class SQSWorkDispatcher:
+    """ SQSWorkDispatcher object that is used to pull work from an SQS queue, and then dispatch it to be
+        executed on a child worker process.
+
+        This has the benefit of the worker process terminating after completion of the work, which will clear all
+        resources used during its execution, and return memory to the operating system. It also allows for periodically
+        monitoring the progress of the worker process from the parent process, and extending the SQS
+        VisibilityTimeout, so that the message is not prematurely terminated and/or moved to the Dead Letter Queue
+
+        Attributes:
+            EXIT_SIGNALS (List[int]): Delineates each of the signals we want to handle, which represent a case where
+                the work being performed is prematurely halting, and the process will exit
+
+
+
+    """
+
+    _logger = logging.getLogger(__name__)
```

Comment:
Was this tested and did it work correctly? While `__name__` is default most loggers in the app are explicitly set to "console". Our logging needs a bit of TLC to differentiate the various logging needs and standardize around them. Until then, we might need this change:


```suggestion
    _logger = logging.getLogger("console")
```